api/scpca_portal/models/library.py

```diff
@@ -175,21 +189,51 @@ def get_metadata(self) -> Dict:
             }
         )
 
-    def get_combined_library_metadata(self) -> List[Dict]:
-        return [
-            self.project.get_metadata() | sample.get_metadata() | self.get_metadata()
-            for sample in self.samples
-        ]
+        return library_metadata
 
-    def get_filtered_data_file_paths(self, download_config: Dict) -> List[Path]:
-        file_extension_to_be_excluded = (
-            common.ANNDATA_EXT
-            if download_config["format"] == Library.FileFormats.SINGLE_CELL_EXPERIMENT
-            else common.SCE_EXT
-        )
+    def get_combined_library_metadata(self) -> List[Dict]:
+        combined_metadatas = []
+        for sample in self.samples.all():
+            metadata = self.project.get_metadata() | sample.get_metadata() | self.get_metadata()
+            # Estimate attributes per modality:
+            #   Single Cell: "sample_cell_count_estimate"
+            #   Single Cell Multiplexed: "sample_cell_estimates"
+            #   Spatial: None
+            if self.modality == Library.Modalities.SPATIAL or self.is_multiplexed:
+                del metadata["sample_cell_count_estimate"]
+            if not self.is_multiplexed:
+                del metadata["sample_cell_estimates"]
+
+            combined_metadatas.append(metadata)
+
+        return combined_metadatas
+
+    def get_download_config_file_paths(self, download_config: Dict) -> List[Path]:
+        """
+        Return all of a library's file paths that are suitable for the passed download config.
+        """
+        omit_suffixes = set()
+        # If metadata_only download, then leave omit suffixes as empty set, else populate
+        if not download_config.get("metadata_only", False):
+            omit_suffixes = set(common.FORMAT_EXTENSIONS.values())
+            # If merged download, keep omit suffixes set full as no library data files are included
+            # If not, remove download config format from omit suffixes set
+            if not download_config.get("includes_merged", False):
+                omit_suffixes.remove(common.FORMAT_EXTENSIONS.get(download_config["format"]))
```

Comment:
```suggestion

        if download_config.get("metadata_only", False):
            return []

        omit_suffixes = set(common.FORMAT_EXTENSIONS.values())

        if not download_config.get("includes_merged", False):
            requested_format = common.FORMAT_EXTENSIONS.get(download_config["format"])
            omit_suffixes.remove(requested_format)
            
```

It would be nice to get rid of the negated check but we still want to remove everything for metadata only download, meaning no files at all.